Marlin/src/module/endstops.cpp

```diff
@@ -293,10 +293,10 @@ void Endstops::not_homing() {
 void Endstops::resync() {
   if (!abort_enabled()) return;     // If endstops/probes are disabled the loop below can hang
 
-  #if ENABLED(ENDSTOP_INTERRUPTS_FEATURE) && !ENDSTOP_NOISE_THRESHOLD
+  #if ENABLED(ENDSTOP_INTERRUPTS_FEATURE)
     update();
   #else
-    safe_delay(2);  // Wait for Temperature ISR (runs at 1KHz)
+    safe_delay(2);  // Wait for Temperature ISR calls update at least 1 time (runs at 1KHz)
```

Comment:
Perhaps the temperature ISR should keep a global counter of how many times it's run, increment at the beginning of the ISR, and you wait until whatever it is has changed? That way you assured that no magic number is needed - the delay won't complete until the ISR runs.

Reply:
That will require at least 1 byte of ram to store the counter and code to increment it...I do not know if it's worth it